### ISSUE TITLE
Make the result of the introspection call available in a header

### ISSUE THREAD

**Issue by djannot** [solo-io/gloo#3767]
**Is your feature request related to a problem? Please describe.**
Gloo allows someone to authenticate with an access token.
An AuthConfig must be created with the `introspectionUrl` set, but the JSON returned by this call isn't available to be consumed by a custom plugin or by the upstream service.

**Describe the solution you'd like**
Add a flag to store the output in a header.
A custom plugin or the upstream service can then access this data.

**Comment by kdorosh** [comment]
this is already available for custom extauth plugins in the request state

do we really need/want the token introspection response in a header?

**Comment by ctomc** [comment]
Testing on 1.5.4 setting up introspection + custom plugin, state field is empty in custom plugin
config
```
apiVersion: enterprise.gloo.solo.io/v1
kind: AuthConfig
metadata:
  name: oidc-hydra
  namespace: gloo-system
spec:
  configs:
    - oauth2:
        accessTokenValidation:
          introspectionUrl: url-to-oath-server
    - pluginAuth:
        name: TokendataPlugin
        pluginFileName: TokendataPlugin.so
        exportedSymbolName: Plugin
```
code of plugin:

```
func (c *TokenDataAuthService) Authorize(ctx context.Context, request *api.AuthorizationRequest) (*api.AuthorizationResponse, error) {
	fmt.Printf("request %+v\n", request)
	fmt.Printf("state %+v\n", state)
...
}

```
results in flowing log:

```
2020-10-29T21:32:06.543919907Z request &{CheckRequest:attributes:<source:<address:<socket_address:<address:"10.198.128.62" port_value:2160 > > > destination:<address:<socket_address:<address:"100.102.230.178" port_value:8080 > > > request:<time:<seconds:1604007126 nanos:538329000 > http:<id:"14466241381529974365" method:"GET" headers:<key:":authority" value:"a89ce06411dbd46cbba448936a137172-95daec75de572c3a.elb.us-east-1.amazonaws.com" > headers:<key:":method" value:"GET" > headers:<key:":path" value:"/httpbin/headers" > headers:<key:"accept" value:"*/*" > headers:<key:"accept-encoding" value:"gzip, deflate, br" > headers:<key:"authorization" value:"Bearer KsHNYcaYTViJB3zZq7v8ceL_PdHT-394DkBbCFu--wk.Kt5i2cQXNIwu3ohJG76Y7nvMAAguptD5RGBecPNv1MA" > headers:<key:"cache-control" value:"no-cache" > headers:<key:"postman-token" value:"c6a74ca8-4230-4bfd-ad58-cfd0a908365b" > headers:<key:"user-agent" value:"PostmanRuntime/7.26.5" > headers:<key:"x-forwarded-proto" value:"http" > headers:<key:"x-request-id" value:"1afc4dc2-f910-4cc0-85a8-6c350cd0f6e7" > path:"/httpbin/headers" host:"xxxxx.elb.us-east-1.amazonaws.com" protocol:"HTTP/1.1" > > context_extensions:<key:"config_id" value:"gloo-system.oidc-hydra" > context_extensions:<key:"source_name" value:"" > context_extensions:<key:"source_type" value:"route" > metadata_context:<> >  State:map[]}
2020-10-29T21:32:06.544160116Z state map[]

```

Is there something I am doing wrong that results in State being empty?

**Comment by kdorosh** [comment]
@ctomc , that makes sense, I finished most of a PR for that today, hoping to get into a release by Monday (probably a beta release before a true 1.5 release of GlooE)

**Comment by kdorosh** [comment]
Reopening until Gloo is released with the fix

**Comment by kdorosh** [comment]
Released in Gloo Enterprise 1.6.0-beta6
